jetty-http/src/main/java/org/eclipse/jetty/http/HttpParser.java

```diff
@@ -1983,4 +1981,56 @@ private IllegalCharacterException(State state,HttpTokens.Token token,ByteBuffer
                 LOG.debug(String.format("Illegal character %s in state=%s for buffer %s",token,state,BufferUtil.toDetailString(buffer)));
         }
     }
+
+    public interface ComplianceViolation
+    {
+        String getName();
+        String getDescription();
+        String getUrl();
+    }
+
+    /**
+     * Section Declarations, used for reporting Violations
+     */
+    enum ViolationSection implements ComplianceViolation
```

Comment:
OK `HttpComplianceSection` was not a great name, but an inner class called `ViolationSection` is no better.    **If** we are going to change the name, then it needs to be to a good one.

Reply:
I do like the interface name `ComplianceViolation` perhaps it is growing on me as a name, but it should be top level an not innerclass of HttpParser.
The main API for configuring should be something like:
```
     HttpCompliance.getRfc2616Builder().allow(ComplianceViolation.SOME_VIOLATION).build();
```